### PR TITLE
Add docs for updated deletion api

### PR DESCRIPTION
**Pull request by MichelHollands**
**What this PR does / why we need it**:

This is the updated documentation for the changes to the deletion api in PR https://github.com/grafana/loki/pull/5481 .

**Checklist**
- [X] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

Signed-off-by: Michel Hollands <michel.hollands@grafana.com>

### CODE REVIEW

File: docs/sources/api/_index.md [hunk 1]
```diff
@@ -1094,3 +1093,21 @@ GET /prometheus/api/v1/alerts
...
+
+Displays a web page with the compactor hash ring status, including the state, healthy and last heartbeat time of each compactor.
+
+### `POST /loki/api/v1/delete`
+
+Create a new delete request. More details can be found in the [logs deletion documentation](../operations/storage/logs-deletion.md#request-log-entry-deletion).
+
+### `GET /loki/api/v1/delete`
+
+List all existing delete requests. More details can be found in the [logs deletion documentation](../operations/storage/logs-deletion.md#list-delete-requests).
```
Comment (MasslessParticle) [inline docs/sources/api/_index.md:1109]: The handler for this only lists the deletes for the userID passed in the `X-Scope-OrgID` header. Is that something we should call out in the docs?

Comment (MichelHollands) [reply docs/sources/api/_index.md]: Yes, that makes sense. The current text could be read as returning every delete request. It's been updated.

File: docs/sources/api/_index.md [hunk 2]
```diff
@@ -1094,3 +1093,21 @@ GET /prometheus/api/v1/alerts
...
+
+Displays a web page with the compactor hash ring status, including the state, healthy and last heartbeat time of each compactor.
+
+### `POST /loki/api/v1/delete`
+
+Create a new delete request. More details can be found in the [logs deletion documentation](../operations/storage/logs-deletion.md#request-log-entry-deletion).
+
+### `GET /loki/api/v1/delete`
+
+List the existing delete requests for the tenant specified in the `x-scope-orgid` header. More details can be found in the [logs deletion documentation](../operations/storage/logs-deletion.md#list-delete-requests).
```
Comment (simonswine) [inline docs/sources/api/_index.md:1109]: This is a bit more inline with what the docs say for rules. Just noticed that this feels a bit more consistent.


```suggestion
List the existing delete requests for the authenticated tenant. More details can be found in the [logs deletion documentation](../operations/storage/logs-deletion.md#list-delete-requests).
```

**Review by simonswine — APPROVED** [review]
Thanks for adding those changes. LGTM

File: docs/sources/operations/storage/logs-deletion.md [hunk 3]
```diff
@@ -28,23 +28,23 @@ The Compactor exposes endpoints to allow for the deletion of log entries from sp
...
-POST /loki/api/admin/delete
-PUT /loki/api/admin/delete
+POST /loki/api/v1/delete
+PUT /loki/api/v1/delete
 ```
 
 Query parameters:
 
-* `match[]=<series_selector>`: Repeated label matcher argument that identifies the streams from which to delete. At least one `match[]` argument must be provided.
+* `query[]=<series_selector>`: query argument that identifies the streams from which to delete. One `query[]` argument must be provided.
```
Comment (sandeepsukhani) [inline docs/sources/operations/storage/logs-deletion.md:37]:
```suggestion
* `query=<series_selector>`: query argument that identifies the streams from which to delete.
```

File: docs/sources/operations/storage/logs-deletion.md [hunk 4]
```diff
@@ -28,23 +28,23 @@ The Compactor exposes endpoints to allow for the deletion of log entries from sp
...
 * `start=<rfc3339 | unix_timestamp>`: A timestamp that identifies the start of the time window within which entries will be deleted. If not specified, defaults to 0, the Unix Epoch time.
 * `end=<rfc3339 | unix_timestamp>`: A timestamp that identifies the end of the time window within which entries will be deleted. If not specified, defaults to the current time.
 
 A 204 response indicates success.
 
-URL encode the `match[]` parameter. This sample form of a cURL command URL encodes `match[]={foo="bar"}`:
+URL encode the `query[]` parameter. This sample form of a cURL command URL encodes `query[]={foo="bar"}`:
```
Comment (sandeepsukhani) [inline docs/sources/operations/storage/logs-deletion.md:43]:
```suggestion
URL encode the `query` parameter. This sample form of a cURL command URL encodes `query={foo="bar"}`:
```

File: docs/sources/operations/storage/logs-deletion.md [hunk 5]
```diff
@@ -28,23 +28,23 @@ The Compactor exposes endpoints to allow for the deletion of log entries from sp
...
 
 ```
-curl -g -X POST \ 
-  'http://127.0.0.1:3100/loki/api/admin/delete?match[]={foo="bar"}&start=1591616227&end=1591619692' \ 
+curl -g -X POST \
+  'http://127.0.0.1:3100/loki/api/v1/delete?query[]={foo="bar"}&start=1591616227&end=1591619692' \
```
Comment (sandeepsukhani) [inline docs/sources/operations/storage/logs-deletion.md:47]:
```suggestion
  'http://127.0.0.1:3100/loki/api/v1/delete?query={foo="bar"}&start=1591616227&end=1591619692' \
```

File: docs/sources/api/_index.md [hunk 6]
```diff
@@ -1094,3 +1093,21 @@ GET /prometheus/api/v1/alerts
...
+
+Displays a web page with the compactor hash ring status, including the state, healthy and last heartbeat time of each compactor.
+
+### `POST /loki/api/v1/delete`
+
+Create a new delete request. More details can be found in the [logs deletion documentation](../operations/storage/logs-deletion.md#request-log-entry-deletion).
+
+### `GET /loki/api/v1/delete`
+
+List the existing delete requests for the authenticated tenant. More details can be found in the [logs deletion documentation](../operations/storage/logs-deletion.md#list-delete-requests).
```
Comment (sandeepsukhani) [inline docs/sources/api/_index.md:1109]: > for the authenticated tenant

All the deletion endpoints perform tenant-specific operations so either we should call it out in all the endpoints or not at all.

**Review by sandeepsukhani — COMMENTED** [review]
Suggested some minor nits but other than that it LGTM.

File: docs/sources/api/_index.md (hunk 6 above)
Comment (MichelHollands) [reply docs/sources/api/_index.md:1109]: It's been added to all delete endpoints.

File: docs/sources/api/_index.md [hunk 7]
```diff
@@ -1094,3 +1093,21 @@ GET /prometheus/api/v1/alerts
 Prometheus-compatible rules endpoint to list all active alerts.
 
 For more information, please check out the Prometheus [alerts](https://prometheus.io/docs/prometheus/latest/querying/api/#alerts) documentation.
+
+## Compactor
+
+### `GET /compactor/ring`
+
+Displays a web page with the compactor hash ring status, including the state, healthy and last heartbeat time of each compactor.
```
Comment (KMiller-Grafana) [inline docs/sources/api/_index.md:1101]:
```suggestion
Displays a web page with the compactor hash ring status, including the state, health, and last heartbeat time of each compactor.
```

File: docs/sources/configuration/_index.md [hunk 8]
```diff
@@ -2026,6 +2026,13 @@ compacts index shards to more performant forms.
 # CLI flag: -boltdb.shipper.compactor.max-compaction-parallelism
 [max_compaction_parallelism: <int> | default = 1]
 
+# Deletion mode.
+# Can be one of "disabled", "whole-stream-deletion", "filter-only" or "filter-and-delete".
+# When set to (the default) value of "whole-stream-deletion" and if
+# retention_enabled is true then the Log Entry Deletion API is available
```
Comment (KMiller-Grafana) [inline docs/sources/configuration/_index.md:2032]:
```suggestion
# Can be one of "disabled", "whole-stream-deletion", "filter-only", or "filter-and-delete".
# When set to the default value of "whole-stream-deletion", and if
# retention_enabled is true, then the log entry deletion API endpoints are available.
```

File: docs/sources/operations/storage/logs-deletion.md [hunk 9]
```diff
@@ -4,10 +4,11 @@ weight: 60
 ---
 # Log Entry Deletion
 
-<span style="background-color:#f3f973;">Log entry deletion is experimental. It is only supported for the BoltDB Shipper index store.</span>
+<span style="background-color:#f3f973;">Log entry deletion is only supported for the BoltDB Shipper index store.</span>
```
Comment (KMiller-Grafana) [inline docs/sources/operations/storage/logs-deletion.md:7]: I strongly suggest that this color only be used for the experimental feature note. I don't think that you need a color at all.  This suggestion removes the highlight color and introduces emphasis with italics.  I hope that others doing reviews will provide their opinions on how to note something like this.  Note that too many "Notes" tend to cause the reader to skip over them, the opposite of what you want.

```suggestion
Log entry deletion is supported _only_ for the BoltDB Shipper index store.
```

File: docs/sources/operations/storage/logs-deletion.md [hunk 10]
```diff
@@ -4,10 +4,11 @@ weight: 60
...
+
+Grafana Loki supports the deletion of log entries from a specified stream.
+Log entries that fall within a specified time window and and match an optional line filter are those that will be deleted.
```
Comment (KMiller-Grafana) [inline docs/sources/operations/storage/logs-deletion.md:10]:
```suggestion
Log entries that fall within a specified time window and match an optional line filter are those that will be deleted.
```

File: docs/sources/operations/storage/logs-deletion.md [hunk 11]
```diff
@@ -17,7 +18,12 @@ Log entry deletion relies on configuration of the custom logs retention workflow
 
 ## Configuration
 
-Enable log entry deletion by setting `retention_enabled` to true in the Compactor's configuration. See the example in [Retention Configuration](../retention#retention-configuration).
+Enable log entry deletion by setting `retention_enabled` to true and `deletion_enabled` to `whole-stream-deletion`, `filter-only` or `filter-and-delete` in the Compactor's configuration. See the example in [Retention Configuration](../retention#retention-configuration).
```
Comment (KMiller-Grafana) [inline docs/sources/operations/storage/logs-deletion.md:21]:
```suggestion
Enable log entry deletion by setting `retention_enabled` to true and `deletion_enabled` to `whole-stream-deletion`, `filter-only`, or `filter-and-delete` in the compactor's configuration. See the example in [Retention configuration](../retention#retention-configuration).
```

File: docs/sources/operations/storage/logs-deletion.md [hunk 12]
```diff
@@ -17,7 +18,12 @@ Log entry deletion relies on configuration of the custom logs retention workflow
...
+
+With `whole-stream-deletion` all the log entries matching the query given in the delete request are removed.
+With `filter-only` log lines matching the query in the delete request are filtered out when querying Loki. They are not removed from the on-disk chunks.
+With `filter-and-delete` log lines matching the query in the delete request are filtered out when querying Loki and they are also removed from the on-disk chunks.
```
Comment (KMiller-Grafana) [inline docs/sources/operations/storage/logs-deletion.md:25]:
```suggestion
With `whole-stream-deletion`, all the log entries matching the query given in the delete request are removed.
With `filter-only`, log lines matching the query in the delete request are filtered out when querying Loki. They are not removed from the on-disk chunks.
With `filter-and-delete`, log lines matching the query in the delete request are filtered out when querying Loki, and they are also removed from the on-disk chunks.
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]

**Review by KMiller-Grafana — APPROVED** [review]
Thanks for the revisions.

**Review by kavirajk — APPROVED** [review]
LGTM :+1:

**Pull request merged**